stream_alert/shared/lookup_tables/driver_dynamodb.py

```diff
@@ -0,0 +1,194 @@
+        # FIXME (derek.wang)
+        pass
+
+    def get(self, key, default=None):
+        self._reload_if_necessary(key)
+
+        return self._dynamo_data.get(key, default)
+
+    def set(self, key, value):
+        # FIXME (derek.wang)
+        pass
+
+    def _reload_if_necessary(self, key):
+        """
+        Uses the "cache_refresh_minutes" option to determine whether or not the current LookupTable
+        should be re-fetched from S3.
+
+        If it needs a reload, this method will appropriately call reload.
+        """
+        last_load_time = self._dynamo_load_times.get(key, 0)
+
+        now = datetime.utcnow()
+        refresh_delta = timedelta(minutes=self._cache_refresh_minutes)
+        needs_refresh = last_load_time + refresh_delta < now if last_load_time else True
+
+        if not needs_refresh:
+            LOGGER.debug(
+                'LookupTable (%s): Key %s does not need refresh. Last refresh: %s; Currently: %s',
+                self.id,
+                key,
+                last_load_time,
+                now
+            )
+            return
+
+        LOGGER.info(
+            'LookupTable (%s): Key %s needs refresh, starting now. Last refresh: %s; Currently: %s',
+            self.id,
+            key,
+            last_load_time,
+            now
+        )
+        self._load(key)
+
+    def _load(self, key):
+        # FIXME (derek.wang)
+        #   Because of the way we explode the key using a delim, there is no way to do last-minute
+        #   casting of sort key into a format OTHER than string. A sort key of 'N' (number) type
+        #   simply will never work...
+        if self._dynamo_db_sort_key:
+            components = key.split(self._key_delimiter, 2)
+            if len(components) != 2:
+                LOGGER.error(
+                    'LookupTable (%s): Invalid key. The requested table requires a sort key, '
+                    'which the provided key \'%s\' does not provide, given the configured '
+                    'delimiter: (%s)',
+                    self.id,
+                    key,
+                    self._key_delimiter
+                )
+                return
+
+            key_schema = {
+                self._dynamo_db_partition_key: components[0],
+                self._dynamo_db_sort_key: components[1],
+            }
+        else:
+            key_schema = {
+                self._dynamo_db_partition_key: key,
+            }
+
+        LOGGER.debug(
+            'LookupTable (%s): Loading key \'%s\' with schema (%s)',
+            self.id,
+            key,
+            json.dumps(key_schema)
+        )
+
+        try:
+            response = self._table.get_item(
+                Key=key_schema,
+
+                # It's not urgently vital to do consistent reads; we accept that for some time we
+                # may get out-of-date reads.
+                ConsistentRead=False,
+
+                # FIXME (derek.wang) Should be off for non-debug mode
```

Comment:
are you saying you only want to do `ReturnConsumedCapacity='TOTAL'` if debug is on? you could use the same "guard" mentioned above if so. then set your arts to `get_item` like so:

```python

args = {
  'Key': key_schema,
  'ConsistentRead': False
}

if LOGGER_DEBUG_ENABLED:
  args['ReturnConsumedCapacity'] = 'TOTAL'

response = self._table.get_item(**args)
```